executor/write_test.go

```diff
@@ -3503,6 +3506,274 @@ func (s *testSuite4) TestWriteListPartitionTableIssue21437(c *C) {
+	tk.MustExec("update t set b=b+1 where a=1")
+	tk.MustQuery("select b from t where a=1").Check(testkit.Rows("2"))
+	tk.MustExec("update t set b=b+1 where a<2")
+	tk.MustQuery("select b from t where a<2").Check(testkit.Rows("3"))
+	tk.MustExec("insert into t values (1, 1) on duplicate key update b=b+1")
+	tk.MustQuery("select b from t where a=1").Check(testkit.Rows("4"))
+}
+
+func (s *testSuite4) TestListPartitionWithAutoIncrement(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("set @@session.tidb_enable_table_partition = 1")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec(`create table t (a bigint key auto_increment, b int) partition by list (a%5) (partition p0 values in (0,1,2), partition p1 values in (3,4));`)
+	tk.MustExec("set @@allow_auto_random_explicit_insert = true")
+	tk.MustExec("replace into t values  (1,1)")
+	result := []string{"1"}
+	for i := 2; i < 100; i++ {
+		sql := fmt.Sprintf("insert into t (b) values (%v)", i)
+		tk.MustExec(sql)
+		result = append(result, strconv.Itoa(i))
+	}
+	tk.MustQuery("select b from t order by b").Check(testkit.Rows(result...))
+	tk.MustExec("update t set b=b+1 where a=1")
+	tk.MustQuery("select b from t where a=1").Check(testkit.Rows("2"))
+	tk.MustExec("update t set b=b+1 where a<2")
+	tk.MustQuery("select b from t where a<2").Check(testkit.Rows("3"))
+	tk.MustExec("insert into t values (1, 1) on duplicate key update b=b+1")
+	tk.MustQuery("select b from t where a=1").Check(testkit.Rows("4"))
+}
+
+func (s *testSuite4) TestListPartitionWithGeneratedColumn(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("use test")
+	// Test for generated column with bigint type.
+	tableDefs := []string{
+		// Test for virtual generated column for list partition
+		`create table t (a bigint, b bigint GENERATED ALWAYS AS (3*a - 2*a) VIRTUAL, index idx(a)) partition by list (5*b - 4*b) (partition p0 values in (1,2,3,4,5), partition p1 values in (6,7,8,9,10));`,
+		// Test for stored generated column for list partition
+		`create table t (a bigint, b bigint GENERATED ALWAYS AS (3*a - 2*a) STORED, index idx(a)) partition by list (5*b - 4*b) (partition p0 values in (1,2,3,4,5), partition p1 values in (6,7,8,9,10));`,
+		// Test for virtual generated column for list columns partition
+		`create table t (a bigint, b bigint GENERATED ALWAYS AS (3*a - 2*a) VIRTUAL, index idx(a)) partition by list columns(b) (partition p0 values in (1,2,3,4,5), partition p1 values in (6,7,8,9,10));`,
+		// Test for stored generated column for list columns partition
+		`create table t (a bigint, b bigint GENERATED ALWAYS AS (3*a - 2*a) STORED, index idx(a)) partition by list columns(b) (partition p0 values in (1,2,3,4,5), partition p1 values in (6,7,8,9,10));`,
+	}
+	for _, tbl := range tableDefs {
+		tk.MustExec("drop table if exists t")
+		tk.MustExec("set @@session.tidb_enable_table_partition = 1")
```

Comment:
we can move `tk.MustExec("set @@session.tidb_enable_table_partition = 1")` out of the loop.

Reply:
Great, done.